### ISSUE TITLE
Clarify bytestring conversion docstrings

### ISSUE THREAD

**Issue by cfcs** [mirage/ocaml-ipaddr#81]
We currently have this:
```ocaml
  (** {3 Bytestring conversion} *)

  (** [of_bytes_exn ipv4_octets] is the address represented
      by [ipv4_octets]. Raises [Parse_error] if [ipv4_octets] is not a
      valid representation of an IPv4 address. *)
  val of_bytes_exn : string -> t

  (** Same as [of_bytes_exn] but returns an option type instead of raising
      an exception. *)
  val of_bytes : string -> t option

  (** Same as [of_bytes_exn] but take an extra paramenter, the offset into
      the bytes for reading. *)
  val of_bytes_raw : string -> int -> t
```

From reading the docstring it was unclear to me that
>  if [ipv4_octets] is not a valid representation of an IPv4 address

actually means (which AFAICT is the only failure mode?):

> if [ipv4_octets] is not exactly 4 bytes long.

It is also misleading that `of_bytes_raw` suggests that it is the same as `of_bytes_exn`; as evidenced below, `of_bytes_raw` does not impose an upper bound on the string length:

```ocaml
  let of_bytes_raw bs o =
    make
      (Char.code bs.[0 + o])
      (Char.code bs.[1 + o])
      (Char.code bs.[2 + o])
      (Char.code bs.[3 + o])

  let of_bytes_exn bs =
    let len = String.length bs in
    if len > 4 then raise (too_much bs);
    if len < 4 then raise (need_more bs);
    of_bytes_raw bs 0
```

Not overly important I guess, but this tripped me during a code review.

**Comment by cfcs** [comment]
I think `of_bytes_raw` is nicer than `of_bytes_exn`, it make the calling code much prettier since you don't need to spray `String.(sub s 2 4)` everywhere.

**Comment by avsm** [comment]
I'd prefer to keep of_bytes_raw and fix the docstrings :)

**Comment by hannesm** [comment]
has been fixed by @avsm finally in #89 AFAICT, please reopen (or PR) if there's anything left to do. I'm fine with keeping the `_raw` functions, sorry for the noise.

**Comment by reynir** [comment]
`Ipaddr.V6.of_octets{,_exn}` suffers from the same issue.
